### ISSUE TITLE
[Bug]: Two instances of Nextcloud Desktop running in parallel

### ISSUE THREAD

**Issue by janvlug** [nextcloud/desktop#4661]
### ⚠️ Before submitting, please verify the following: ⚠️

- [X] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github (I've searched it).
- [X] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

Lately, I see regularly that there are two instances of Nextcloud Desktop running after first logging in. I see two Nextcloud Desktop icons:
![image](https://user-images.githubusercontent.com/5903208/174964625-7df62dd1-0075-4b8c-8c46-38dc25bec9cd.png)
And ps shows this:
![image](https://user-images.githubusercontent.com/5903208/174964844-e4857888-1826-4e45-a15d-b730beef6ce0.png)

### Steps to reproduce

1. Start the computer
2. Wait for the system to be booted
3. Log in
Note that two instances of Nextcloud Desktop are started

### Expected behavior

Only one instance of Nextcloud Desktop running at the same time

### Which files are affected by this bug

n.a.

### Operating system

Linux

### Which version of the operating system you are running.

Fedora 35

### Package

Distro package manager

### Nextcloud Server version

22.2.6

### Nextcloud Desktop Client version

3.5.1

### Is this bug present after an update or on a fresh install?

Updated from a minor version (ex. 3.4.2 to 3.4.4)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### Are you using an external user-backend?

- [X] Default internal user-backend
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Additional info

For (still unknown to me) reasons, my system is slow on disk access, and starting applications takes a relative long time.

**Comment by janvlug** [comment]
I noticed now that I have two .desktop files for Nextcloud desktop in .config/autostart/
![image](https://user-images.githubusercontent.com/5903208/174969359-905c0513-22dc-4ca5-b66b-212e615c19a9.png)

**Comment by janvlug** [comment]
After making changes to the Nextcloud Desktop settings, I notice that the com.nextcloud.desktopclient.nextcloud.desktop file is the one that is being modified:
![image](https://user-images.githubusercontent.com/5903208/174970420-cd83ba8b-8f9d-46c7-bd57-c7cfc750ed84.png)

**Comment by mgallien** [comment]
looks like the second desktop file is dating back from release 3.0.0 (or maybe even older but I cannot be sure)
the difference in name might be the result of a packaging issue at the time. Vendor (including Linux distributions) may change it during the build process.
given how old this is, my recommendation is really to delete the one named `Nextcloud.desktop` and checks if everything is now working fine

**Comment by janvlug** [comment]
I confirm that the removing one of the files solved the issue. I removed Nextcloud.desktop, because that was the oldest one.

**Comment by janvlug** [comment]
Today I noticed two instances of Nextcloud running again on my Fedora desktop:

![image](https://github.com/nextcloud/desktop/assets/5903208/c167895e-ade9-40ff-a0be-ac25fac16126)

```
ls -sla ~/.config/autostart/*extcloud*
4 -rw-r--r--. 1 jan jan 244 Nov  3 14:49 /home/jan/.config/autostart/com.nextcloud.desktopclient.nextcloud.desktop
4 -rw-r--r--. 1 jan jan 244 Apr 14 12:22 /home/jan/.config/autostart/Nextcloud.desktop
4 -rw-rw-r--. 1 jan jan 202 May  6  2019 /home/jan/.config/autostart/Nextcloud.desktop.double_bak
```

So it looks like this issue resurfaced. For now I will move the `com.nextcloud.desktopclient.nextcloud.desktop` aside, because that is the oldest.

Is there a risk in running two Nextcloud instances in parallel? Should this somehow be detected by Nextcloud Desktop?

Should this issue be re-opened?

**Comment by janvlug** [comment]
See also:

- https://bugzilla.redhat.com/show_bug.cgi?id=2280905
- https://help.nextcloud.com/t/doubled-nc-client-instances-in-linux/183270

So I think this issue should be re-opened, @camilasan.

**Comment by ChristophWurst** [comment]
Same on Arch Linux

```
-rw-r--r--   1 christoph christoph  244 Apr 29 07:07 Nextcloud.desktop
-rw-r--r--   1 christoph christoph  244 Oct 29  2023 com.nextcloud.desktopclient.nextcloud.desktop
-rw-r--r--   1 christoph christoph 9406 Aug 14  2018 nextcloud.desktop
```

**Comment by mgallien** [comment]
@janvlug it seems like our method to prevent the same client to run twice and detect an already running one may fail in some circumstances
we could indeed revisit this issue

**Comment by janvlug** [comment]
FYI: I did not see this issue any more lately, which of course is no proof that it does not exist any more.

**Comment by FabD86** [comment]
@janvlug @mgallien this problem is still present:
![Image](https://github.com/user-attachments/assets/4418492e-7ae1-4e4f-8634-3ebdd25b33d1)

**Comment by JHSheridan** [comment]
Same issue here

![Image](https://github.com/user-attachments/assets/57a10476-c4e3-4727-8362-cd43b5d39777)

**Comment by ChristophWurst** [comment]
```
$ ps ax | grep nextcloud
  48471 ?        SLl    0:00 /usr/bin/nextcloud
  48697 ?        Sl     0:00 /usr/bin/nextcloud
  48947 ?        Sl     0:00 /usr/bin/nextcloud
  48993 ?        Sl     0:00 /usr/bin/nextcloud
```

while I can't get the client to switch to foreground mode. I can't "see" any of the running clients due to https://github.com/nextcloud/desktop/issues/5771

**Comment by ChristophWurst** [comment]
> Same on Arch Linux
> 
> ```
> -rw-r--r--   1 christoph christoph  244 Apr 29 07:07 Nextcloud.desktop
> -rw-r--r--   1 christoph christoph  244 Oct 29  2023 com.nextcloud.desktopclient.nextcloud.desktop
> -rw-r--r--   1 christoph christoph 9406 Aug 14  2018 nextcloud.desktop
> ```

Contents of these files:

.config/autostart/com.nextcloud.desktopclient.nextcloud.desktop

```
[Desktop Entry]
Name=Nextcloud
GenericName=File Synchronizer
Exec="/usr/bin/nextcloud" --background
Terminal=false
Icon=Nextcloud
Categories=Network
Type=Application
StartupNotify=false
X-GNOME-Autostart-enabled=true
X-GNOME-Autostart-Delay=10
```

.config/autostart/nextcloud.desktop

```
[Desktop Entry]
Categories=Utility;X-SuSE-SyncUtility;
Type=Application
Exec=nextcloud
Name=Nextcloud desktop sync client 
Comment=Nextcloud desktop synchronization client
GenericName=Folder Sync
Icon=Nextcloud
Keywords=Nextcloud;syncing;file;sharing;
X-GNOME-Autostart-Delay=3

# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations


# Translations
Comment[oc]=Nextcloud sincronizacion del client
GenericName[oc]=Dorsièr de Sincronizacion
Name[oc]=Nextcloud sincronizacion del client
Icon[oc]=Nextcloud
Comment[ar]=Nextcloud زبون مزامنة مكتبي
GenericName[ar]=مزامنة المجلد
Name[ar]=Nextcloud زبون مزامنة مكتبي
Icon[ar]=Nextcloud
Comment[bg_BG]=Nextcloud клиент за десктоп синхронизация
GenericName[bg_BG]=Синхронизиране на папката
Name[bg_BG]=Nextcloud клиент десктоп синхронизация
Icon[bg_BG]=Nextcloud
Comment[ca]=Client de sincronització d'escriptori Nextcloud
GenericName[ca]=Sincronització de carpetes
Name[ca]=Client de sincronització d'escriptori Nextcloud
Icon[ca]=Nextcloud
Comment[da]=Nextcloud skrivebordsklient til synkronisering
GenericName[da]=Mappesynkronisering
Name[da]=Nextcloud skrivebordsklient til synk
Icon[da]=Nextcloud
Comment[de]=Nextcloud Desktop-Synchronisationsclient
GenericName[de]=Ordner-Synchronisation
Name[de]=Nextcloud Desktop-Synchronisationsclient
Icon[de]=Nextcloud
Comment[ja_JP]=Nextcloud デスクトップ同期クライアント
GenericName[ja_JP]=フォルダー同期
Name[ja_JP]=Nextcloud デスクトップ同期クライアント
Icon[ja_JP]=Nextcloud
Comment[el]=@ΟΝΟΜΑ_ΕΦΑΡΜΟΓΗΣ@ συγχρονισμός επιφάνειας εργασίας πελάτη
GenericName[el]=Συγχρονισμός φακέλου
Name[el]=@ΟΝΟΜΑ_ΕΦΑΡΜΟΓΗΣ@  συγχρονισμός επιφάνειας εργασίας πελάτη
Icon[el]=Nextcloud
Comment[en_GB]=Nextcloud desktop synchronisation client
GenericName[en_GB]=Folder Sync
Name[en_GB]=Nextcloud desktop sync client 
Icon[en_GB]=Nextcloud
Comment[es]=Nextcloud cliente de sincronización de escritorio
GenericName[es]=Sincronización de carpeta
Name[es]=Nextcloud cliente de sincronización de escritorio
Icon[es]=Nextcloud
Comment[de_DE]=Nextcloud Desktop-Synchronisationsclient
GenericName[de_DE]=Ordner-Synchronisation
Name[de_DE]=Nextcloud Desktop-Synchronisationsclient
Icon[de_DE]=Nextcloud
Comment[eu]=Nextcloud mahaigaineko sinkronizazio bezeroa
GenericName[eu]=Karpetaren sinkronizazioa
Name[eu]=Nextcloud mahaigaineko sinkronizazio bezeroa
Icon[eu]=Nextcloud
GenericName[fa]=همسان سازی پوشه‌ها
Name[fa]=nextcloud نسخه‌ی همسان سازی مشتری
Icon[fa]=Nextcloud
Comment[fr]=Synchronisez vos dossiers avec un serveur Nextcloud
GenericName[fr]=Synchronisation de dossier
Name[fr]=Client de synchronisation Nextcloud
Icon[fr]=Nextcloud
Comment[gl]=Nextcloud cliente de sincronización para escritorio
GenericName[gl]=Sincronizar Cartafol
Name[gl]=Nextcloud cliente de sincronización para escritorio
Icon[gl]=Nextcloud
Comment[he]=Nextcloud לקוח סנכון שולחן עבודה
GenericName[he]=סנכון תיקייה
Name[he]=Nextcloud לקוח סנכרון שולחן עבודה 
Icon[he]=Nextcloud
Comment[ia]=Nextcloud cliente de synchronisation pro scriptorio
GenericName[ia]=Synchronisar Dossier 
Name[ia]=Nextcloud cliente de synchronisation pro scriptorio
Icon[ia]=Nextcloud
Comment[id]=Klien sinkronisasi desktop Nextcloud
GenericName[id]=Folder Sync
Name[id]=Klien sync desktop Nextcloud
Icon[id]=Nextcloud
Comment[is]=Nextcloud skjáborðsforrit samstillingar
GenericName[is]=Samstilling möppu
Name[is]=Nextcloud skjáborðsforrit samstillingar
Icon[is]=Nextcloud
Comment[it]=Client di sincronizzazione del desktop di Nextcloud
GenericName[it]=Sincronizzazione cartella
Name[it]=Client di sincronizzazione del desktop di Nextcloud
Icon[it]=Nextcloud
Comment[ko]=Nextcloud 데스크톱 동기화 클라이언트
GenericName[ko]=폴더 동기화
Name[ko]=Nextcloud 데스크톱 동기화 클라이언트
Comment[hu_HU]=Nextcloud asztali szinkronizációs kliens
GenericName[hu_HU]=Könyvtár szinkronizálás
Name[hu_HU]=Nextcloud asztali szinkr. kliens
Icon[hu_HU]=Nextcloud
Comment[nl]=Nextcloud desktop synchronisatie client
GenericName[nl]=Mappen sync
Name[nl]=Nextcloud desktop sync client 
Icon[nl]=Nextcloud
Comment[et_EE]=Nextcloud sünkroonimise klient töölauale
GenericName[et_EE]=Kaustade sünkroonimine
Name[et_EE]=Nextcloud sünkroonimise klient töölauale
Icon[et_EE]=Nextcloud
Comment[pl]=Nextcloud klient synchronizacji dla komputerów stacjonarnych
GenericName[pl]=Folder Synchronizacji
Name[pl]=Nextcloud klient synchronizacji dla komputerów stacjonarnych
Icon[pl]=Nextcloud
Comment[pt_BR]=Nextcloud cliente de sincronização do computador
GenericName[pt_BR]=Sincronização de Pasta
Name[pt_BR]=Nextcloud cliente de sincronização de desktop
Icon[pt_BR]=Nextcloud
Comment[cs_CZ]=Nextcloud počítačový synchronizační klient
GenericName[cs_CZ]=Synchronizace adresáře
Name[cs_CZ]=Nextcloud počítačový synchronizační klient
Icon[cs_CZ]=Nextcloud
Comment[ru]=Настольный клиент синхронизации Nextcloud 
GenericName[ru]=Синхронизация каталогов
Name[ru]=Настольный клиент синхронизации Nextcloud
Icon[ru]=Nextcloud
Comment[sl]=Nextcloud ‒ Program za usklajevanje datotek z namizjem
GenericName[sl]=Usklajevanje map
Name[sl]=Nextcloud ‒ Program za usklajevanje datotek z namizjem
Icon[sl]=Nextcloud
Comment[sq]=Klient njëkohësimesh Nextcloud për desktop
GenericName[sq]=Njëkohësim Dosjesh
Name[sq]=Klient njëkohësimesh Nextcloud për desktop
Icon[sq]=Nextcloud
Comment[fi_FI]=Nextcloud työpöytäsynkronointisovellus
GenericName[fi_FI]=Kansion synkronointi
Name[fi_FI]=Nextcloud työpöytäsynkronointisovellus
Icon[fi_FI]=Nextcloud
Comment[sv]=Nextcloud desktop synkroniseringsklient
GenericName[sv]=Mappsynk
Name[sv]=Nextcloud desktop synk-klient
Icon[sv]=Nextcloud
Comment[tr]=Nextcloud masaüstü eşitleme istemcisi
GenericName[tr]=Dosya Eşitleme
Name[tr]=Nextcloud masaüstü eşitleme istemcisi
Icon[tr]=Nextcloud
Comment[uk]=Настільний клієнт синхронізації Nextcloud
GenericName[uk]=Синхронізація теки
Name[uk]=Настільний клієнт синхронізації Nextcloud
Icon[uk]=Nextcloud
Comment[ro]=Nextcloud client de sincronizare pe desktop
GenericName[ro]=Sincronizare director
Name[ro]=Nextcloud client de sincronizare pe desktop
Icon[ro]=Nextcloud
Comment[zh_CN]=Nextcloud 桌面同步客户端
GenericName[zh_CN]=文件夹同步
Name[zh_CN]=Nextcloud 桌面同步客户端
Icon[zh_CN]=Nextcloud
Comment[zh_HK]=桌面版同步客户端
GenericName[zh_TW]=資料夾同步
Comment[es_AR]=Cliente de sincronización para escritorio Nextcloud 
GenericName[es_AR]=Sincronización de directorio
Name[es_AR]=Cliente de sincronización para escritorio Nextcloud 
Icon[es_AR]=Nextcloud
Comment[lt_LT]=Nextcloud darbalaukio sinchronizavimo programa
GenericName[lt_LT]=Katalogo sinchnorizacija
Name[lt_LT]=Nextcloud darbalaukio programa
Icon[lt_LT]=Nextcloud
Comment[th_TH]=Nextcloud ไคลเอนต์ประสานข้อมูลเดสก์ท็อป
GenericName[th_TH]=ประสานข้อมูลโฟลเดอร์
Name[th_TH]= Nextcloud ไคลเอนต์ประสานข้อมูลเดสก์ท็อป
Icon[th_TH]=Nextcloud
Comment[es_MX]=Cliente de escritorio para  sincronziación de Nextcloud
GenericName[es_MX]=Sincronización de Carpetas
Name[es_MX]=Cliente de escritorio para  sincronziación de Nextcloud
Icon[es_MX]=Nextcloud
Comment[nb_NO]=Nextcloud skrivebordssynkroniseringsklient
GenericName[nb_NO]=Mappesynkronisering
Name[nb_NO]=Nextcloud skrivebordssynkroniseringsklient 
Icon[nb_NO]=Nextcloud
Comment[nn_NO]=Nextcloud klient for å synkronisera frå skrivebord
GenericName[nn_NO]=Mappe synkronisering
Name[nn_NO]=Nextcloud klient for å synkronisera frå skrivebord
Icon[nn_NO]=Nextcloud
Comment[pt_PT]=Nextcloud - Cliente de Sincronização para PC
GenericName[pt_PT]=Sincronizar Pasta
Name[pt_PT]=Nextcloud - Cliente de Sincronização para PC
Icon[pt_PT]=Nextcloud
Icon[km]=Nextcloud
Comment[lb]=Nextcloud Desktop Synchronisatioun Client
GenericName[lb]=Dossier Dync
Name[lb]=Nextcloud Desktop Sync Client
Icon[lb]=Nextcloud
```

.config/autostart/Nextcloud.desktop
```
[Desktop Entry]
Name=Nextcloud
GenericName=File Synchronizer
Exec="/usr/bin/nextcloud" --background
Terminal=false
Icon=Nextcloud
Categories=Network
Type=Application
StartupNotify=false
X-GNOME-Autostart-enabled=true
X-GNOME-Autostart-Delay=10
```

learning: **.config/autostart/com.nextcloud.desktopclient.nextcloud.desktop and .config/autostart/Nextcloud.desktop are identical**.

**Comment by olberger** [comment]
FWIW, related Debian bug report: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1095002

**Comment by olberger** [comment]
You may check which file should be the current one by stopping all instances of Nextcloud Desktop, then removing all instances of ~/.config/autostart/*extcloud.desktop, then restarting nextcloud, and check which files are created/removed when going to the preferences and checking/unchecking the parameter "launch on startup" (approx translation. haven't checked other than my locale's).

On my Debian, I had to get rid of com.nextcloud.desktopclient.nextcloud.desktop, since Nextcloud.desktop is the one that gets recreated if I'm changing the settings.

**Comment by janvlug** [comment]
I did not see this issue for quite a while, but today I was hit by this issue again: 

<img width="144" height="45" alt="Image" src="https://github.com/user-attachments/assets/673459fc-1b6b-49f3-b37c-95f799ff3cc7" />

Running Fedora Linux.

nextcloud-client.x86_64                                 3.16.3-2.fc42

I got strange warnings about previously ignored files, and was wondering what was wrong. Then I noted that there were two Nextcloud instances running again.

**Comment by janvlug** [comment]
Today this happened again:

<img width="160" height="52" alt="Image" src="https://github.com/user-attachments/assets/fd727368-3bd5-42fb-a936-9c312c226831" />

Maybe after clicking on a notification by Nextcloud on Fedora.

I saw unexpected notifications about added external storage. I went to investigate and noticed that there were two Nextcloud desktop instances running.

nextcloud-client.x86_64                                 3.17.4-1.fc43

**Comment by janvlug** [comment]
After that I had the two instances running yesterday, I noticed now that all large folders stopped synchronizing. This is pretty scary for me.

**Comment by janvlug** [comment]
Fedora ABRT detected a Nextcloud client crash around the time that the two instances were running. This crash has been automatically reported [here](https://bugzilla.redhat.com/show_bug.cgi?id=2440380).

**Comment by dixDel** [comment]
Same problem on MacOS. Nextcloud is configured to open at login. I regularly (not always) see 2 instances when opening the computer. The icons are unresponsive to clicks.

<img width="75" height="36" alt="Image" src="https://github.com/user-attachments/assets/34e62f6a-137b-4263-a165-d7e150ea42f9" />

**Comment by artfulrobot** [comment]
I've had this numerous times over several years. It causes immense problems because when there are two running, one of them decides unilaterally to sync everything, not just the things you have selected; so then you can end up downloading TBs of stuff, potentially crashing your laptop, and/or causing other things to stop syncing properly.

I've experienced it on Linux (Fedora, Debian, Ubuntu). This morning on Fedora, where NC is installed through the package archives. `nextcloud-client.x86_64          3.17.4-1.fc43` 

I only have one .desktop file, at `com.nextcloud.desktopclient.nextcloud.desktop` 

EDIT: on reading https://github.com/nextcloud/desktop/issues/4661#issuecomment-3216590993, I see in .config/autostart:

```
Dec 23 23:38 Nextcloud.desktop
Jun  3  2024 com.nextcloud.desktopclient.nextcloud.desktop
```

But I note that as of writing this, it's March 2026; both these files appear to have been here a long time (both "birth" times in stat are from 2024).

Could it be that when the package manager installs a new version, it starts it up, ignoring a check for whether the old one was running?

**Comment by melroy89** [comment]
Today I also got two instances running suddenly. I got Nextcloud (v33) installed via PPA  on Linux Mint `33.0.0-20260326.054717.659bb51b22-1.0~noble1`.

https://launchpad.net/~nextcloud-devs/+archive/ubuntu/client

So one start-up happens due to `~/.config/autostart` (how it used to work). And the other instance is because of systemd: `/usr/lib/systemd/user/com.nextcloud.desktopclient.nextcloud.service`

```sh
❯ ps -aux | grep nextcloud
melroy      5307  0.7  0.2 3530984 280716 ?      SLsl 15:44   0:02 /usr/bin/nextcloud --background
melroy      5591  0.4  0.2 3293704 269964 ?      SLl  15:44   0:01 /usr/bin/nextcloud --background
```

<img width="99" height="46" alt="Image" src="https://github.com/user-attachments/assets/2e71dd75-b8bf-4b0f-af6d-29025134e0d1" />

I only see one Nextcloud startup app in the list:

<img width="734" height="459" alt="Image" src="https://github.com/user-attachments/assets/c005063f-7b5c-42cc-87a9-67197b0fa698" />

---

I'm pretty sure this is a  regression issue in NC v33 in deb for sure.

Because the [latest deb](https://launchpad.net/~nextcloud-devs/+archive/ubuntu/client) installs a `systemd` startup as well: `/usr/lib/systemd/user/com.nextcloud.desktopclient.nextcloud.service`:

```ini
[Unit]
Description=Nextcloud desktop user service
Wants=network-online.target

[Service]
ExecStart=/usr/bin/nextcloud --background
Restart=on-failure
RestartSec=5s
NoNewPrivileges=true
RestrictRealtime=true

[Install]
WantedBy=default.target
```

> Could it be that when the package manager installs a new version, it starts it up, ignoring a check for whether the old one was running?

@artfulrobot I think the latest package in the package managers now also installs systemd ON TOP OF autostart.

Meaning Nextcloud desktop now triggers both autostart and systemd!  @mgallien Please resolve this asap, This is really a regression you want to solve quickly.

**Comment by janvlug** [comment]
It seems that several times the two Nextcloud Desktop instances running at the same time were caused by auto starting or issues with Linux packaging. Would it be a solution if Nextcloud Desktop (better) detects and prevents being started more than once (on Linux)? Should I open a new issue for that?

**Comment by melroy89** [comment]
See also https://github.com/nextcloud/desktop/issues/9729 where the NC developer did respond to.

**Comment by GregorKopka** [comment]
> Because the [latest deb](https://launchpad.net/~nextcloud-devs/+archive/ubuntu/client) installs a `systemd` startup as well: `/usr/lib/systemd/user/com.nextcloud.desktopclient.nextcloud.service`

Why would the .deb install a service aimed at default.target (or [graphical.target](https://github.com/nextcloud/desktop/commit/e40106c551b0e81fbe24226e5bc7937cae9f8779)) **for something that should run in a user context and with user privileges** (to get the configuration and to not be able to do `rm / -R`) ?

**Comment by Jerome-Herbinet** [comment]
Please note that this particular bug is a **real pain**. I hope this will be fixed ASAP (**urgent hotfix version**).

**Comment by pwaring** [comment]
I'm also experiencing this bug on Xubuntu 24.04 with the Nextcloud PPA. I am not sure what triggers it as it will happen intermittently, but once it starts it will go on until I intervene by deleting the duplicate entry from the Application Autostart tab of the Session and Startup app. It has happened again recently and I noticed that I'd upgraded the Nextcloud client, so I don't know if that is part of the issue. On this occasion it hasn't duplicated all my files, but it has done so in the past and I had to work out some magic with the find command to delete and remove all the duplicates.

**Comment by johannesh19** [comment]
Xubuntu 24.04 from the PPA here as well. After startup, there are two instances of the client running – sometimes immediately, sometimes a second instance starts after 5–15 (?) minutes only.
If I `pkill nextcloud*` I can restart the client manually. In this case, there's usually only one running until shutdown. However, two or three times during the last weeks, after a while (~ 2–3 hours?) a second instance started itself with no apparent trigger.

**Comment by FabriceOE** [comment]
There are two instances of service start in the systemd, so the app start twice and nothing works
(Linux mint 22.1, Ubuntu 24.04)

the old one (from somewhere in 2025) is
"com.nextcloud.desktopclient.nextcloud.service" at /var/lib/systemd/deb-systemd-user-helper-enabled/defaut.target.wants

the new one (from my latest update)
"com.nextcloud.desktopclient.nextcloud.service" at /var/lib/systemd/deb-systemd-user-helper-enabled/graphical.target.wants

-->you need to delete the first one

for dev: please also check if instance is running, if not do not launch twice...

**Comment by RalfJung** [comment]
That's just two dependencies on the same service. That would not on its own lead to the service being started twice.

**Comment by FabriceOE** [comment]
On my system, the fix above stopped the  "dual instance" (2 icons in status bar, freeze, interference)
--> now I only have a single icon (and everything seems to run as usual)

**Comment by RalfJung** [comment]
That's not how systemd works, so there must be some other issue going on.

**Comment by unkilbeegtoo** [comment]
Linux Mint 22.3 here.

What seems to be happening *here* is that `Nextcloud.desktop` appears to be created in `.config/autostart`.

In *addition*, there is a `user/default.target.wants/com.nextcloud.desktopclient.nextcloud.service` under systemd.  Deleting either of these fixes the problem.

Until the missing one is re-created.  I don't know who is doing that -- it's either the running nextcloud instance or during an update.  But after some short period of time, the second icon appears in my toolbar again, and both startup methods are back in place.  I kill the extra one at each boot until I get annoyed enough to track down the extra file and delete it.

**Comment by brainchild0** [comment]
It seems that the selection of applications for auto start is configurable per user, within the desktop environment, and therefore not bound to the contents of any system package.

As the application is now managed by the system supervisor (e.g. `systemd`), it is obviously unnecessary that it also be included in the selection of the applications for auto start by the desktop environment.

Those asking for auto start to be removed from the distributed package seem to asking the impossible. Regardless, it is trivial to reconfigure one's own desktop environment such as not to start the application, which is rather managed by the supervisor.

**Comment by unkilbeegtoo** [comment]
There are several issues.  One is that updates seem to add both startup methods even after the user has disabled one.  You will find yourself dealing with this problem again and again, even after you have disabled the autostart in your desktop environment.  It comes  back.

Another problem is that the `systemd` startup is a bad idea for multiuser systems.  It means that each user has Nextcloud start up, even if they don't have a Nextcloud account.  Machines used by multiple users *do* exist.  If a user does not have an account, they will be faced with the Nextcloud sign-in pop-up on every login (which annoyingly is pinned to the middle of the desktop and can be hard to dismiss.)

It makes sense for the startup to be managed on an individual basis, not started up for each user on a system-wide basis.  The application runs for the individual -- it should be manageable by the individual user.  Handling it with `systemd` prevents individual choice.

But that's probably an issue for a different bug report.

**Comment by kafeenstra** [comment]
seeing the same issue here on linux mint 22.3:
```
$ ls -l .config/autostart/Nextcloud.desktop 
-rw-rw-r-- 1 feenstra feenstra 244 Sep  3 18:02 .config/autostart/Nextcloud.desktop
$ ls -l /etc/systemd/user/default.target.wants/com.nextcloud.desktopclient.nextcloud.service
lrwxrwxrwx 1 root root 67 Feb 27  2026 /etc/systemd/user/default.target.wants/com.nextcloud.desktopclient.nextcloud.service -> /usr/lib/systemd/user/com.nextcloud.desktopclient.nextcloud.service
```
I've had this before in mid-august, removed the systemd one, which got re-created with the late august update. Only surfaced now after reboot.
Should this be a separate bug report?
